Makefile

```diff
@@ -584,7 +584,10 @@ $(LIBROACH_DIR)/Makefile: $(C_DEPS_DIR)/libroach-rebuild $(BOOTSTRAP_TARGET)
 	mkdir -p $(LIBROACH_DIR)
 	@# NOTE: If you change the CMake flags below, bump the version in
 	@# $(C_DEPS_DIR)/libroach-rebuild. See above for rationale.
-	cd $(LIBROACH_DIR) && cmake $(CMAKE_FLAGS) $(LIBROACH_SRC_DIR) -DCMAKE_BUILD_TYPE=Release
+	cd $(LIBROACH_DIR) && cmake $(CMAKE_FLAGS) $(LIBROACH_SRC_DIR) -DCMAKE_BUILD_TYPE=Release \
+		-DPROTOBUF_LIB=$(PROTOBUF_DIR)/libprotobuf.a -DROCKSDB_LIB=$(ROCKSDB_DIR)/librocksdb.a \
+		$(if $(USE_STDMALLOC),,-DJEMALLOC_LIB=$(JEMALLOC_DIR)/lib/libjemalloc.a) \
```

Comment:
I don't think this works quite right; `make build; make build TYPE=msan` is going to do the wrong thing because CMake will reuse the configuration from the first time around.

If it's all the same to you, I think we should just drop the conditional and mandate for now that C++ unit tests link against jemalloc. We can always revisit if/when we hook up CI to run msan builds.

Reply:
Fair enough. Removed.